enhancements/cluster-logging/loki-storage.md

```diff
@@ -0,0 +1,628 @@
+    //
+    // +optional
+    Ingester *LokiComponentSpec `json:"ingester,omitempty"`
+
+    // Querier defines the querier component spec.
+    //
+    // +optional
+    Querier *LokiComponentSpec `json:"querier,omitempty"`
+
+    // QueryFrontend defines the query frontend component spec.
+    //
+    // +optional
+    QueryFrontend *LokiComponentSpec `json:"queryFrontend,omitempty"`
+}
+```
+
+### Status
+
+The `LokiStackStatus` is quite spartan and declares the most important conditions of the Loki deployment that cannot be expressed by metrics. A managed Loki deployment can be either in state `Ready` or `Degraded`. In summary conditions under which the Loki deployment is `Degraded` are:
+
+1. The secret resource to access the object storage is either missing or invalid, e.g. missing `bucketnames` or malformed `endpoint`.
+2. The selected redudancy policy is not available for the selected scale out size, e.g. `SingleRedundancy` and `1x.small` are incompatible.
+
+If the configuration given in a `LokiStack` custom resource results in a `Degraded` cluster, **no** Loki deployment will be created.
+
+```go
+// LokiStackConditionType deifnes the type of condition types of a Loki deployment.
+type LokiStackConditionType string
+
+const (
+    // ConditionReady defines the condition that all components in the Loki deployment are ready.
+    ConditionReady LokiStackConditionType = "Ready"
+
+    // ConditionDegraded defines the condition that some or all components in the Loki deployment
+    // are degraded or the cluster cannot connect to object storage.
+    ConditionDegraded LokiStackConditionType = "Degraded"
+)
+
+// LokiStackConditionReason defines the type for valid reasons of a Loki deployment conditions.
+type LokiStackConditionReason string
+
+const (
+    // ReasonMissingObjectStorageSecret when the required secret to store logs to object
+    // storage is missing.
+    ReasonMissingObjectStorageSecret LokiStackConditionReason = "MissingObjectStorageSecret"
+
+    // ReasonInvalidObjectStorageSecret when the format of the secret is invalid.
+    ReasonInvalidObjectStorageSecret LokiStackConditionReason = "InvalidObjectStorageSecret"
+
+    // ReasonInvalidRedundancyConfiguration when the configurated redudancy is not valid
+    // with the select cluster size.
+    ReasonInvalidRedundancyConfiguration LokiStackConditionReason = "InvalidRedundancyConfiguration"
+)
+
+// LokiStackStatus defines the observed state of LokiStack
+type LokiStackStatus struct {
+    // Conditions of the Loki deployment health.
+    //
+    // +optional
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+```
+
+### clusterlogging_types.go
+
+The integration of the `LokiStack` CRD into `ClusterLogging/v1` custom resources is following the present pattern to first into the latter.
+
+```go
+type LogStoreType string
+
+const (
+    LogStoreTypeElasticsearch LogStoreType = "elasticsearch"
+
+    LogStoreTypeLoki  LogStoreType = "Loki"
```

Comment:
Lowercase "loki", in keeping with the rest of "types" in the CL CRD,  type: "elasticsearch", type: "kibana", type: "fluentd"?